packages/sanity/src/core/bundles/components/dialog/BundleDetailsDialog.tsx

```diff
@@ -0,0 +1,122 @@
+import {ArrowRightIcon} from '@sanity/icons'
+import {Box, Button, Dialog, Flex} from '@sanity/ui'
+import {type FormEvent, useCallback, useState} from 'react'
+
+import {type BundleDocument} from '../../../store/bundles/types'
+import {useBundleOperations} from '../../../store/bundles/useBundleOperations'
+import {usePerspective} from '../../hooks/usePerspective'
+import {BundleForm} from './BundleForm'
+
+interface BundleDetailsDialogProps {
+  onCancel: () => void
+  onSubmit: () => void
+  bundle?: BundleDocument
+}
+
+export function BundleDetailsDialog(props: BundleDetailsDialogProps): JSX.Element {
+  const {onCancel, onSubmit, bundle} = props
+  const {createBundle, updateBundle} = useBundleOperations()
+  const [hasErrors, setHasErrors] = useState(false)
+
+  const [value, setValue] = useState<Partial<BundleDocument>>(() => {
```

Comment:
using a function to set default state depending on whether existing bundle value already exists